File: example_interfaces/testable/output/python/src/testableipc/property.py
from pydantic import BaseModel
from .interface_types import *


class TestableInitialPropertyValues(BaseModel):

    read_write_integer: int
    read_write_integer_version: int = 0

    read_only_integer: int
    read_only_integer_version: int = 0

    read_write_optional_integer: Optional[int]
    read_write_optional_integer_version: int = 0

    read_write_two_integers: ReadWriteTwoIntegersProperty
    read_write_two_integers_version: int = 0

    read_only_string: str
    read_only_string_version: int = 0

    read_write_string: str
    read_write_string_version: int = 0

    read_write_optional_string: Optional[str]
    read_write_optional_string_version: int = 0

    read_write_two_strings: ReadWriteTwoStringsProperty
    read_write_two_strings_version: int = 0

    read_write_struct: AllTypes
    read_write_struct_version: int = 0

    read_write_optional_struct: AllTypes
    read_write_optional_struct_version: int = 0

    read_write_two_structs: ReadWriteTwoStructsProperty
    read_write_two_structs_version: int = 0

    read_only_enum: Numbers
    read_only_enum_version: int = 0

    read_write_enum: Numbers
    read_write_enum_version: int = 0

    read_write_optional_enum: Optional[Numbers]
    read_write_optional_enum_version: int = 0

    read_write_two_enums: ReadWriteTwoEnumsProperty
    read_write_two_enums_version: int = 0

    read_write_datetime: datetime
    read_write_datetime_version: int = 0

    read_write_optional_datetime: Optional[datetime]
    read_write_optional_datetime_version: int = 0

    read_write_two_datetimes: ReadWriteTwoDatetimesProperty
    read_write_two_datetimes_version: int = 0

    read_write_duration: timedelta
    read_write_duration_version: int = 0

    read_write_optional_duration: Optional[timedelta]
    read_write_optional_duration_version: int = 0

    read_write_two_durations: ReadWriteTwoDurationsProperty
    read_write_two_durations_version: int = 0

    read_write_binary: bytes
    read_write_binary_version: int = 0

    read_write_optional_binary: bytes
    read_write_optional_binary_version: int = 0

    read_write_two_binaries: ReadWriteTwoBinariesProperty
    read_write_two_binaries_version: int = 0

    read_write_list_of_strings: List[str]
    read_write_list_of_strings_version: int = 0

    read_write_lists: ReadWriteListsProperty
    read_write_lists_version: int = 0

File: example_interfaces/testable/output/rust/src/property.rs
#[allow(unused_imports)]
use crate::payloads::*;
use derive_builder::Builder;

#[derive(Clone, Builder, Debug)]
pub struct TestableInitialPropertyValues {
    pub read_write_integer: i32,
    pub read_write_integer_version: u32,

    pub read_only_integer: i32,
    pub read_only_integer_version: u32,

    #[builder(default = "None")]
    pub read_write_optional_integer: Option<i32>,
    pub read_write_optional_integer_version: u32,

    pub read_write_two_integers: ReadWriteTwoIntegersProperty,
    pub read_write_two_integers_version: u32,

    pub read_only_string: String,
    pub read_only_string_version: u32,

    pub read_write_string: String,
    pub read_write_string_version: u32,

    #[builder(default = "None")]
    pub read_write_optional_string: Option<String>,
    pub read_write_optional_string_version: u32,

    pub read_write_two_strings: ReadWriteTwoStringsProperty,
    pub read_write_two_strings_version: u32,

    pub read_write_struct: AllTypes,
    pub read_write_struct_version: u32,

    #[builder(default = "None")]
    pub read_write_optional_struct: Option<AllTypes>,
    pub read_write_optional_struct_version: u32,

    pub read_write_two_structs: ReadWriteTwoStructsProperty,
    pub read_write_two_structs_version: u32,

    pub read_only_enum: Numbers,
    pub read_only_enum_version: u32,

    pub read_write_enum: Numbers,
    pub read_write_enum_version: u32,

    #[builder(default = "None")]
    pub read_write_optional_enum: Option<Numbers>,
    pub read_write_optional_enum_version: u32,

    pub read_write_two_enums: ReadWriteTwoEnumsProperty,
    pub read_write_two_enums_version: u32,

    pub read_write_datetime: chrono::DateTime<chrono::Utc>,
    pub read_write_datetime_version: u32,

    #[builder(default = "None")]
    pub read_write_optional_datetime: Option<chrono::DateTime<chrono::Utc>>,
    pub read_write_optional_datetime_version: u32,

    pub read_write_two_datetimes: ReadWriteTwoDatetimesProperty,
    pub read_write_two_datetimes_version: u32,

    pub read_write_duration: chrono::Duration,
    pub read_write_duration_version: u32,

    #[builder(default = "None")]
    pub read_write_optional_duration: Option<chrono::Duration>,
    pub read_write_optional_duration_version: u32,

    pub read_write_two_durations: ReadWriteTwoDurationsProperty,
    pub read_write_two_durations_version: u32,

    pub read_write_binary: Vec<u8>,
    pub read_write_binary_version: u32,

    #[builder(default = "None")]
    pub read_write_optional_binary: Option<Vec<u8>>,
    pub read_write_optional_binary_version: u32,

    pub read_write_two_binaries: ReadWriteTwoBinariesProperty,
    pub read_write_two_binaries_version: u32,

    pub read_write_list_of_strings: Vec<String>,
    pub read_write_list_of_strings_version: u32,

    pub read_write_lists: ReadWriteListsProperty,
    pub read_write_lists_version: u32,
}
